blocks/procedures.js

```diff
@@ -128,19 +127,27 @@ Blockly.Blocks['procedures_defnoreturn'] = {
    */
   domToMutation: function(xmlElement) {
     this.arguments_ = [];
+    this.varIds_ = [];
+    this.paramIds_ = [];
     this.argumentVarModels_ = [];
     for (var i = 0, childNode; (childNode = xmlElement.childNodes[i]); i++) {
       if (childNode.nodeName.toLowerCase() == 'arg') {
         var varName = childNode.getAttribute('name');
         var varId = childNode.getAttribute('varid') || childNode.getAttribute('varId');
+        this.varIds_.push(varId);
         this.arguments_.push(varName);
         var variable = Blockly.Variables.getOrCreateVariablePackage(
             this.workspace, varId, varName, '');
         if (variable != null) {
           this.argumentVarModels_.push(variable);
         } else {
-          console.log('Failed to create a variable with name ' + varName + ', ignoring.');
+          console.log('Failed to create a variable with name ' +
```

Comment:
There's a pre-existing problem here, but it's worth addressing if you're changing this code: If no variable can be created we still add to most of the arrays (`arguments_`, `varIds_`, `paramIds_`) but not to `argumentVarModels_`.  That means they're out of sync permanently.

Reply:
Good idea!